### PR TITLE
Copy haproxy-extra certificates directory if exists

### PR DESCRIPTION
**Pull request by Wasaac**
Change-Id: I3e2fc25df69987adf5a17d3b909f2904c1f9ba06

### CODE REVIEW

File: ansible/roles/haproxy/tasks/config.yml [hunk 1]
```diff
@@ -125,6 +125,29 @@
   notify:
     - Restart haproxy container
 
+- name: Checking if haproxy-extra certificates dir exists
+  run_once: true
+  local_action: stat path={{ haproxy_extra_cert_directory }}
```
Comment (jovial) [inline ansible/roles/haproxy/tasks/config.yml:130]: I seem to recall `delegate_to: localhost` being preferred over local action in kolla-ansible: https://github.com/openstack/kolla-ansible/commit/25e98f9047165908cbfb6b12c993de179f04b56b

**Review by jovial — DISMISSED** [review]

File: ansible/roles/haproxy/tasks/config.yml [hunk 2]
```diff
@@ -125,6 +125,31 @@
...
+  register: haproxy_extra_cert_dir
+  changed_when: false
+
+- name: Copying over haproxy-extra certificates
+  vars:
+    service: "{{ haproxy_services['haproxy'] }}"
+  copy:
+    src: "{{ haproxy_extra_cert_directory }}/"
+    dest: "{{ node_config_directory }}/haproxy/certs/"
+    mode: "0660"
```
Comment (dougszumski) [inline ansible/roles/haproxy/tasks/config.yml:142]: Why 0660 / 0770?

**Review by dougszumski — APPROVED** [review]

File: ansible/roles/haproxy/tasks/config.yml (hunk 2 above)
Comment (Wasaac) [reply ansible/roles/haproxy/tasks/config.yml:142]: Certs are given u+g rw perms, same as haproxy.pem.
Directories are given u+g rwx, the same perms as `{{ node_config_directory }}/haproxy` and other config dirs.
Arguably these could be tightened but are at worst no less secure than `{{ node_config_directory }}/`.

**Comment by markgoddard** [comment]
I don't see these patches upstream.

**Pull request merged**